New: profile page

RujaMarcello opened the pull request.

RujaMarcello committed:
New: profile page

Ovyzz commented on backend/api/user.js:
```diff
@@ -1,24 +1,28 @@
 const { verifyToken } = require("../middleware/auth");
-
+const { isEmailValid } = require("../helper.js");
+const jwt = require("jsonwebtoken");
```
You don need to use jwt outside validator

Ovyzz commented on backend/api/user.js:
```diff
@@ -31,6 +35,25 @@ router.get("/user/current", verifyToken, async (req, res) => {
+  const emailIsValid = await isEmailValid(data.email);
+
+  const token = req.headers.token;
+  const decriptedToken = jwt.verify(token, process.env.TOKEN_KEY);
```
You need to find another aproach for that

Ovyzz commented on backend/api/user.js:
```diff
@@ -31,6 +35,25 @@ router.get("/user/current", verifyToken, async (req, res) => {
+
+  const token = req.headers.token;
+  const decriptedToken = jwt.verify(token, process.env.TOKEN_KEY);
+  if (decriptedToken.email != data.email) {
```
```suggestion
  if (decriptedToken.email !== data.email) {
```

Ovyzz commented on frontend/src/components/header-content/index.module.scss:
```diff
@@ -0,0 +1,9 @@
+
```
You have one space here
```suggestion
```

Ovyzz commented on frontend/src/components/header-content/index.tsx:
```diff
@@ -0,0 +1,36 @@
+import { useNavigate } from 'react-router-dom';
+
+import styles from './index.module.scss';
+interface EmptyLayoutProps {
```
Why you use this interface here?

Ovyzz commented on frontend/src/components/profile-form/index.module.scss:
```diff
@@ -0,0 +1,67 @@
+
+}
+
+
```
```suggestion
```

Ovyzz commented on frontend/src/components/profile-form/index.tsx:
```diff
@@ -0,0 +1,130 @@
+
+interface ProfileFormProps {
+  onSubmit: (form: ProfileFormState) => void;
+  loading: boolean | undefined;
```
why loading can be undefined?

Ovyzz commented on frontend/src/pages/dashboard/profile/index.tsx:
```diff
@@ -0,0 +1,36 @@
+          profilePictureUrl: form.profilePictureUrl,
+        },
+      });
+      toast.success('Updated');
```
That toast dosen t work every time

RujaMarcello committed:
Refactor, Update, Fix

Refactor:  code if email is the same with previous one.
Refactor:  empty llines in index.module.scss in header-content and profile-form.
Fix: tost is data was succesfully updated.
Update: backend database.sql.
Fix: loading state is only boolean.

RujaMarcello committed:
Update

Update: format scss files

Ovyzz commented on backend/.env.example:
```diff
@@ -0,0 +1,2 @@
+REACT_DATABASE_PASSWORD = Your DB password
+TOKEN_KEY = Your token key
```
For what do you use TOKEN_KEY?
